plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -193,6 +200,64 @@ def _state_deleted(self, want, have):
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+        commands = []
+
+        diff1 = get_diff(have, want, TEST_KEYS, is_skeleton=True)
+        if diff1:
+            requests = self.get_delete_all_mclag_domain_request(have)
+            if len(requests) > 0:
+                send_requests(self._module, requests)
+            else:
+                commands = []
+
+        exist_mclag_facts = self.get_mclag_facts()
+        have_new = exist_mclag_facts
+        diff_new = get_diff(want, have_new)
+
+        if diff_new:
+            requests = self.get_create_mclag_request(want, diff_new)
+            if len(requests) > 0:
+                commands = update_states(diff, "replaced")
+        return commands, requests
+
+    def _state_overridden(self, want, have, diff):
+        """ The command generator when state is overridden
+        :param want: the desired configuration as a dictionary
+        :param obj_in_have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        diff1 = get_diff(have, want)
```

Comment:
The diff was provided as an input parameter, so this extra invocation should be removed. If there is any content in the input diff, all configuration for this resource module should be deleted, then the commands in "want" should be applied. Please re-work this section accordingly.

Reply:
Diff calculated between "have" and "want". If there is no difference obtained in both cases, no configuration is applied. If there is a difference, then "delete all" is handled. Refer line #246

Reply:
Thanks for explaining. This is a reasonable way of accomplishing the intended check.

I will mark this issue "Resolved."

Reply:
Oh. You already did. (Please leave resolution of code review issues for the reviewer whenever possible.)